src/state/opamFormatUpgrade.ml

```diff
@@ -1210,6 +1219,24 @@ let as_necessary requested_lock global_lock root config =
     (* Deadlock until one is killed in interactive mode, but abort in batch *)
     if OpamStd.Sys.tty_out then None else Some true
   in
+  let config =
+    let config, missing_switches = remove_missing_switches root config in
+    let local, global =
+      List.partition OpamSwitch.is_external missing_switches
+    in
+    let tostr sws =
+      OpamStd.Format.pretty_list
+        (List.map (OpamSwitch.to_string @> OpamConsole.colorise `bold) sws)
+    in
+    let switches = function [_] -> "" | _ -> "es" in
+    if global <> [] then
+      OpamConsole.warning "Broken global switch%s %s"
```

Comment:
The message in the reftests looks strange because it creates "Broken global switch missing". These switches are also removed, though, so could the sentence be in the same form as for local switches (I agree it's a warning - the user can reasonably delete local switches by accident, but manually deleting a global switch is naughty!).

"Removing references to broken global switch%s %s as it/they no longer exists/exist"?

Reply:
I intentionally didn't use the same wording of 'removal' because this is printed even in light upgrades (non blocking), users won't understand why a switch will be removed when they just want to install a package for example. But yes, better have two different sentences : 
* light: _Broken global switch%s %s_ , or use the same as in https://github.com/OCamlPro/opam/blob/master/src/state/opamSwitchState.ml#L30 _Removing local switch %s as it no longer exists_
* hard: _Removing references to broken global switch%s %s as it/they no longer exists/exist_ or the same as local ones
?